core/src/main/java/org/apache/struts2/interceptor/I18nInterceptor.java

```diff
@@ -281,15 +282,13 @@ public Locale find() {
 
         @Override
         public Locale store(ActionInvocation invocation, Locale locale) {
-            HttpSession session = ServletActionContext.getRequest().getSession(false);
+            Map<String, Object> session = invocation.getInvocationContext().getSession();
```

Comment:
Why not to call `HttpSession session = ServletActionContext.getRequest().getSession();` directly? The the `if` below can be dropped.

Reply:
I think the `invocation.getInvocationContext().getSession()` can be `null` in some cases, can't it? And we need it to put locale.

Reply:
ah, right :)